Source/SuperLinq/CopyTo.cs

```diff
@@ -157,15 +175,31 @@ public static void CopyTo<TSource>(this IEnumerable<TSource> source, IList<TSour
 		}
 		else if (list is TSource[] array)
 		{
-			if (TryGetCollectionCount(source, out var n)
-				&& n > array.Length)
+			if (source is ICollection<TSource> coll)
```

Comment:
This "ICollection but not Array" case is the only one that can't have the rest of the logic forwarded to the "Span target" version. After this block (or at least for the else) forward to the Span overload for clarity or create a private method to hold the common logic.

Reply:
private method makes sense. Done.